tls/s2n_connection.h

```diff
@@ -120,6 +99,17 @@ struct s2n_connection {
     /* The PRF needs some storage elements to work with */
     union s2n_prf_working_space prf_space;
 
+    /* Whether to use client_cert_auth_type stored in s2n_config or in this s2n_connection.
+     *
+     * By default the s2n_connection will defer to s2n_config->client_cert_auth_type on whether or not to use Client Auth.
+     * But users can override Client Auth at the connection level using s2n_connection_set_client_auth_type() without mutating
+     * s2n_config since s2n_config can be shared between multiple s2n_connections. */
+    uint8_t client_cert_auth_type_overridden;
```

Comment:
Is there a way to infer this instead of having to explicitly set it and set the options on the connection?